### PR TITLE
Fixes for Android

### PR DESCRIPTION
**Pull request by dkurt**

### CODE REVIEW

**Comment by JacekDanecki** [comment]
Can you rebase and clean your PR, so unit tests pass on external CI?

**Comment by dkurt** [comment]
@JacekDanecki, Done, thank you!

File: runtime/os_interface/linux/drm_neo.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ int Drm::ioctl(unsigned long request, void *arg) {
     int ret;
     SYSTEM_ENTER();
     do {
-        ret = ::ioctl(fd, request, arg);
+        ret = ::ioctl(fd, static_cast<unsigned int>(request), arg);
```
Comment (JacekDanecki) [inline runtime/os_interface/linux/drm_neo.cpp]: Why do you need this cast?

Comment (dkurt) [reply runtime/os_interface/linux/drm_neo.cpp]: Android NDK compiler says cannot convert from unsigned long to unsigned int.

Comment (alalek) [reply runtime/os_interface/linux/drm_neo.cpp]: Linux `ioctl` signature is:

> int ioctl(int fd, unsigned long request, ...);

from here: http://man7.org/linux/man-pages/man2/ioctl.2.html

Android's declaration looks like:

> int ioctl(int __fd, int __request, ...);

https://android.googlesource.com/platform/bionic/+/master/libc/include/bits/ioctl.h#43

---

On x86 Linux-based systems both `int` and `long` are 32-bit.

Perhaps `#if defined(__ANDROID__)` guard can help here.

Comment (dkurt) [reply runtime/os_interface/linux/drm_neo.cpp]: Thanks! Replaced.

**Comment by JacekDanecki** [comment]
Can you run clang-format on files you've changed, otherwise it will fail our automatic checks during verification.

File: offline_compiler/utilities/linux/safety_guard_linux.h [hunk 2]
```diff
@@ -28,6 +30,7 @@ class SafetyGuardLinux {
     }
 
     static void sigAction(int sig_num, siginfo_t *info, void *ucontext) {
+#if !defined(__ANDROID__)
```
Comment (JablonskiMateusz) [inline offline_compiler/utilities/linux/safety_guard_linux.h]: please don't put code under `if defined` . If android needs other implementation, it should be placed in separated file, i.e. one file for linux, one file for android - cmake chooses correct file based on `ANDROID` flag.

File: runtime/os_interface/linux/drm_neo.cpp (hunk 1 above)
Comment (JablonskiMateusz) [reply runtime/os_interface/linux/drm_neo.cpp]: can you perform the separation at cmake level? I mean one file for linux, one file for android - cmake chooses correct file.

File: offline_compiler/utilities/linux/safety_guard_linux.h (hunk 2 above)
Comment (dkurt) [reply offline_compiler/utilities/linux/safety_guard_linux.h]: What about `drm_neo.cpp`? There are about 40 files there.

**Comment by dkurt** [comment]
@JacekDanecki, the only failed build this one: https://travis-ci.org/intel/compute-runtime/jobs/604502095?utm_medium=notification&utm_source=github_status due to timeout. I think code style is passed.

**Comment by JacekDanecki** [comment]
I've restarted Travis build, but there is still problem with formatting, see below diff. There are 2 spaces in your code. We don't run clang-format tests on external CI yet. There are more verification tests in our internal builds.
```
--- core/unit_tests/helpers/memory_management.cpp	(before formatting)
+++ core/unit_tests/helpers/memory_management.cpp	(after formatting)
@@ -314,7 +314,7 @@
         printf("for detailed stack information turn on captureCallStacks in memory_management.h\n");
     }
     if (event.frames > 0) {
-#if defined(__linux__)  && !defined(__ANDROID__)
+#if defined(__linux__) && !defined(__ANDROID__)
         char **bt = backtrace_symbols(event.callstack, event.frames);
         char *demangled;
         int status;
```
Can you split files as @JablonskiMateusz suggested?

File: offline_compiler/utilities/linux/safety_guard_linux.h (hunk 2 above)
Comment (JablonskiMateusz) [reply offline_compiler/utilities/linux/safety_guard_linux.h]: If Linux version of any class doesn't meet Android requirements, then you should create Android specific implementation in separated file. Currently we have such separation between Linux and Windows.

Please see:
`runtime\dll\linux\create_drm_memory_manager.cpp`
`runtime\dll\windows\create_wddm_memory_manager.cpp`
They both implement single method `std::unique_ptr<MemoryManager> MemoryManager::createMemoryManager(ExecutionEnvironment &executionEnvironment)`

**Pull request closed without merging**